esmvaltool/diag_scripts/weighting/weighted_temperature_map.py

```diff
@@ -0,0 +1,188 @@
+"""Implementation of a mapplot for the climwip weighting scheme.
+
+Lukas Brunner et al. section 2.4
+https://iopscience.iop.org/article/10.1088/1748-9326/ab492f
+"""
+import logging
+import os
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+from cartopy.mpl.ticker import LongitudeFormatter, LatitudeFormatter
+import numpy as np
+import xarray as xr
+from climwip import log_provenance, read_model_data
+from climwip import get_diagnostic_filename, get_plot_filename
+
+from esmvaltool.diag_scripts.shared import run_diagnostic
+from esmvaltool.diag_scripts.weighting.plot_utilities import (
+    read_weights,
+    read_metadata,
+    weighted_quantile,
+)
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def mapplot(dataarray, cfg, title_pattern, filename_part, ancestors, cmap=None, center=None):
+    """Visualize weighted temperature."""
+    period = '{start_year}-{end_year}'.format(**read_metadata(cfg)['tas'][0])
+    if 'tas_reference' in read_metadata(cfg).keys():
+        meta = read_metadata(cfg)['tas_reference']
+        period = 'change: {} minus {start_year}-{end_year}'.format(period, **meta[0])
+    metric = cfg['model_aggregation']
+    if isinstance(metric, int):
+        metric = f'{metric}perc'
+    proj = ccrs.PlateCarree(central_longitude=0)
+    fig, ax = plt.subplots(subplot_kw={'projection': proj})
+
+    dataarray.plot.pcolormesh(
+        ax=ax,
+        transform=ccrs.PlateCarree(),
+        levels=9,
+        robust=True,
+        extend='both',
+        cmap=cmap,
+        center=center,
+        # colorbar size often does not fit nicely
+        # https://stackoverflow.com/questions/18195758/set-matplotlib-colorbar-size-to-match-graph
+        # cbar_kwargs={'fraction': .021}
+    )
+
+    ax.coastlines()
+    lons = dataarray.lon.values
+    lats = dataarray.lat.values
+    longitude_formatter = LongitudeFormatter()
+    latitude_formatter = LatitudeFormatter()
+    if 'xticks' in cfg.keys():
+        ax.set_xticks(cfg['xticks'])
+    else:
+        ax.set_xticks(np.arange(np.floor(lons.min()), np.ceil(lons.max()), 10), crs=proj)
```

Comment:
Just a suggestion (not tested):
```suggestion
    default_xticks = np.arange(np.floor(lons.min()), np.ceil(lons.max()), 10)
    xticks = cfg.get('xticks', default_xticks)
    ax.set_xticks(xticks, crs=proj)
```